lib/absinthe/phase/document/execution/resolution.ex

```diff
@@ -48,11 +48,27 @@ defmodule Absinthe.Phase.Document.Execution.Resolution do
 
     exec = plugins |> run_callbacks(:before_resolution, exec, run_callbacks?)
 
-    {result, exec} =
+    common =
+      Map.take(exec, [:adapter, :context, :acc, :root_value, :schema, :fragments, :fields_cache])
+
+    res =
+      %Absinthe.Resolution{
+        path: nil,
+        source: nil,
+        parent_type: nil,
+        middleware: nil,
+        definition: nil,
+        arguments: nil
+      }
+      |> Map.merge(common)
```

Comment:
```elixir
%{adapter: adapter, context: context, acc: acc, root_value: root_value, schema: schema, fragments: fragments, fields_cache: fields_cache} = exec

%Absinthe.Resolution{
  path: nil,
  source: nil,
  parent_type: nil,
  middleware: nil,
  definition: nil,
  arguments: nil,
  adapter: adapter, 
  context: context, 
  acc: acc, 
  root_value: root_value, 
  schema: schema, 
  fragments: fragments, 
  fields_cache: fields_cache
}
```
explicit maps should be even slightly more performant, though quite verbose in here

Reply:
Yeah, this allocation happens just once per request though, the biggest win was https://github.com/absinthe-graphql/absinthe/pull/636/files#diff-9e88b6c0fafa4e1d518da81f1c8f16a8L195 which is called for every field, potentially thousands of times.

Reply:
so a resolution is still created on every request? I was noticing how big that struct is and wondered if it would slow things down to be created on every request when you are trying to handle dozens per second on an instance. googling led me here :)

Reply:
Hi @atomkirk dozens of requests per second should be trivially achieved on a single instance, the presence of large maps is not going to be your bottleneck.

Reply:
I think it probably also greatly reduces allocations when the vast majority of the keys are atoms. I would assume atoms are shared across all of the vm's processes.

Reply:
That is correct, structs benefit from the VM's constant heap allocation. Not just the keys, but the entire "key structure" part of the map is allocated once when the application boots, and then referenced.